src/Components/Calculator.js

```diff
@@ -1,35 +1,22 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import Row from './RowCalcul';
 import calculate from './logic/calculator';
 import '../css/Calc.css';
 
-class Calculator extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      dataObj: {
-        total: null,
-        next: null,
-        operation: null,
-      },
-    };
-    this.updateDataObj = this.updateDataObj.bind(this);
-  }
-
-  componentDidUpdate() {
-    this.updateScreen();
-  }
+export default function Calculator() {
+  const [dataObj, setDataObj] = useState({
+    total: null,
+    next: null,
+    operation: null,
+  });
 
-  updateDataObj(event) {
-    const val = event.target.textContent;
-    this.setState((prevState) => ({
-      dataObj: { ...calculate(prevState.dataObj, val) },
-    }));
+  function updateDataObj(event) {
+    const value = event.target.textContent;
+    setDataObj((prevDataObj) => calculate(prevDataObj, value));
   }
 
-  updateScreen() {
+  function updateScreen() {
     const screen = document.querySelector('.result-screen');
-    const { dataObj } = this.state;
     if (dataObj.next && !dataObj.total) {
       screen.value = dataObj.next;
     } else if (!dataObj.next && dataObj.total) {
```

Comment:
- [ ]  Kindly refrain from using `document.querySelector` in react. When you try to modify the element that was retrieved using querySelector, you're no longer taking the advantage of React virtual DOM which is the main advantage of using react. Even if you want to only `retrieve` info from the real dom, you should use the `useRef` hook instead. Please implement the display logic using state instead since state should be the only thing that causes what's displayed in the DOM to change.

Reply:
Thank you